src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -224,7 +224,6 @@ internal JsonConverter GetConverterInternal(Type typeToConvert)
 
                 if (s_defaultSimpleConverters.TryGetValue(typeToConvert, out JsonConverter? foundConverter))
                 {
-                    Debug.Assert(foundConverter != null);
```

Comment:
nit: why remove this? nullability of `TryGetValue` doesn't guarantee non-null when true

Reply:
It is because we have the declared the value type of the dictionary as non-nullable. The compiler accepts that `foundConverter` is non-null in this branch so the assertion is redundant.